src/infer_request.cc

```diff
@@ -443,6 +458,13 @@ InferenceRequest::RespondIfError(
           std::move(response), TRITONSERVER_RESPONSE_COMPLETE_FINAL, status),
       (request->LogRequest() + "failed to send error response").c_str());
 
+  request->ReportStatistics(
```

Comment:
While calling `ReportStatistics` may be easy, is it the right thing to call? Will it have any adverse side effects? Will it mess up the `request::trace_`? Please take a look over this part of the flow and see if it makes more sense to extract out a helper or something. It's possible this is correct, but just want a second look at it. Keep in mind the possibility of `ReportStatistics` also getting called twice now with the call from `TRITONBACKEND_ModelInstanceReportStatistics` as well, and that may be undesirable.

Reply:
Separated out the helper to avoid interacting with request::trace_
Added new func 
void ReportErrorStatistics(